custom_components/google_home/api.py

```diff
@@ -236,73 +237,108 @@ async def delete_alarm_or_timer(
         """Deletes a timer or alarm.
         Can also delete multiple if a list is provided (Not implemented yet)."""
 
+        data = {"ids": [item_to_delete]}
+
+        item_type = item_to_delete.split("/")[0]
+
+        _LOGGER.debug(
+            "Deleting %s from Google Home device %s - Raw data: %s",
+            item_type,
+            device.name,
+            data,
+        )
+
+        response = await self.post(
+            endpoint=API_ENDPOINT_DELETE, data=json.dumps(data), device=device
+        )
+
+        if response:
+            if "success" in response:
+                if response["success"]:
+                    _LOGGER.debug(
+                        "Successfully deleted %s for %s",
+                        item_type,
+                        device.name,
+                    )
+                else:
+                    _LOGGER.error(
+                        "Couldn't delete %s for %s - %s",
+                        item_type,
+                        device.name,
+                        response,
+                    )
+            else:
+                _LOGGER.error(
+                    (
+                        "Failed to get a confirmation that the %s"
+                        "was deleted for device %s. "
+                        "Received = %s"
+                    ),
+                    item_type,
+                    device.name,
+                    response,
+                )
+
+    async def reboot_google_device(self, device: GoogleHomeDevice) -> None:
+        """Reboots a Google Home device if it supports this."""
+
+        # "now" means reboot and "fdr" means factory reset (Not implemented).
+        data = {"params": "now"}
+
+        _LOGGER.debug(
+            "Trying to reboot Google Home device %s",
+            device.name,
+        )
+
+        response = await self.post(
+            endpoint=API_ENDPOINT_REBOOT, data=json.dumps(data), device=device
+        )
+
+        if response:
+            # It will return true even if the device does not support rebooting.
+            _LOGGER.info(
+                "Successfully asked %s to reboot.",
+                device.name,
+            )
+
+    async def post(  # type: ignore[misc]
+        self, endpoint: str, data: str, device: GoogleHomeDevice
+    ) -> Any:
```

Comment:
I think this should work. Should be enough precision and better that `Any`.
```suggestion
    ) -> Optional[Dict[str, str]]:
```

Reply:
I will test it 😊

Reply:
Worked! Thanks for the help